src/main/python/extraction/controllers.py

```diff
@@ -15,67 +17,146 @@
+    data = {"success": False}
+
+    # check to see if this is a post request
+    if request.method == "POST":
+        # check to see if an image was uploaded
+        if request.files.get("image", None) is not None:
+            # grab the uploaded image
+            image = _grab_image(stream=request.files["image"])
+
+        # otherwise, assume that a URL was passed in
+        else:
+            # grab the URL from the request
+            url = request.post.get("url", None)
+
+            # if the URL is None, then return an error
+            if url is None:
+                data["error"] = "No URL provided."
+                return jsonify(data)
+
+            # load the image and convert
+            image = _grab_image(url=url)
+    # Call open CV commands here with the extracted image
     print(image)
     face = "jklanskjcbniugciuhncoiaksc6565"
-    return jsonify({"Extracted Face": face})
+    return jsonify({"extracted_face": face})
 
 
 @extract.route('/extractAll', methods=['POST'])
 def extract_all():
     """
     ----------------------------------------------------------------------
-    Author: Nicolai van Niekerk
+    Author(s): Nicolai van Niekerk, Stephan Nell
     ----------------------------------------------------------------------
     Sample function to extract face and text from image received
     ----------------------------------------------------------------------
     URL: http://localhost:5000/extractAll
     ----------------------------------------------------------------------
     """
-    image = request.files.get("image")
-    print(image)
-    data = {
-        "Surname": "Doe",
-        "Names": "John Jane",
-        "Sex": "M",
-        "Nationality": "RSA",
-        "Identity Number": "6944585228083",
-        "Date of Birth": "06-05-1996",
-        "Country of Birth": "RSA",
-        "Status": "Citizen",
-        "Face": "jklanskjcbniugciuhncoiaksc6565"
-    }
-    return jsonify({"Extracted data": data})
+    # initialize the data dictionary to be returned by the request
+    data = {"success": False}
+    # check to see if this is a post request
+    if request.method == "POST":
+        # check to see if an image was uploaded
+        if request.files.get("image", None) is not None:
+            # grab the uploaded image
+            image = _grab_image(stream=request.files["image"])
+
+        # otherwise, assume that a URL was passed in
+        else:
+            # grab the URL from the request
+            url = request.post.get("url", None)
+
+            # if the URL is None, then return an error
+            if url is None:
+                data["error"] = "No URL provided."
+                return jsonify(data)
+
+            # load the image and convert
+            image = _grab_image(url=url)
+        # Call open CV commands here with the extracted image
+        print(image)
+        data.update({"surname": "Nell", "names": "Stephan Jack", "Sex": "M", "nationality": "RSA",
```

Comment:
Sex -> sex

Split ook hierdie hele dictionary dat elke property op sy eie lyn is in plaas van 'n variable number properties per lyn, m.a.w. in plaas van
```python
{"surname": "Nell", "names": "Stephan Jack", "sex": "M", "nationality": "RSA",...
```
doen jy 
```python
{
    "surname": "Nell",
    "names": "Stephan Jack",
    "sex": "M",
    "nationality": "RSA",
    ...
}
```
Dit is sodat ons netjies en consistent kan wees. 'n Dict is OF 1 lyn lank, OF 1 lyn per property, afhangende van hoe baie properties daar is. Dit maak dinge ook net meer leesbaar.